cli/azd/cmd/init.go

```diff
@@ -324,5 +328,18 @@ func (i *initAction) Run(ctx context.Context) error {
 		return fmt.Errorf("saving default environment: %w", err)
 	}
 
+	// If the configuration is empty, set default subscription & location
+	// This will be the case for first run experience
+	if !i.accountManager.HasDefaults() {
+		_, err = i.accountManager.SetDefaultSubscription(ctx, env.GetSubscriptionId())
+		if err != nil {
+			log.Printf("failed setting default subscription. %s\n", err.Error())
+		}
+		_, err = i.accountManager.SetDefaultLocation(ctx, env.GetSubscriptionId(), env.GetLocation())
+		if err != nil {
+			log.Printf("failed setting default location. %s\n", err.Error())
+		}
+	}
```

Comment:
Reply to [previous thread](https://github.com/Azure/azure-dev/pull/967#discussion_r1003869599):

I don't really think of azd init as the forced starting point for the user. A user can run azd provision and azd deploy, and never touch azd init. For example, they're using their own repository, or cloning via github, or forking via github.

To me, the "first-run" experience means: when the user selects a subscription and location from the prompt and does not yet have the values defined in the config file. Thus, I would save the default values if not present when a selection is made in the location prompt. I think this also avoids this weird side-effect of azd init persisting the defaulted values when it doesn't really use the subscription/location values or the config.

Reply:
Going to keep it in `init` for the time being.  Will wait to hear if there are any other opinions on this.  I think setting the defaults during `azd init` would be more expected vs the side effect of doing it during a selection prompt.

Reply:
FWIW, I agree with @weikanglim here.  We've tried to build the system such that an explicit `azd init` is not required for cases like he points out.  Like him, I think it's more understandable if we just write these values back to config the first time we prompt for them, regardless of the command we are running.